service/resource/deployment/deployment.go

```diff
@@ -19,49 +19,71 @@ func getDeploymentNames() []string {
 	}
 }
 
-func (r Resource) newMainDeployment(cluster providerv1alpha1.AzureConfig) (Deployment, error) {
-	certs, err := r.certsSearcher.SearchCluster(key.ClusterID(cluster))
+func (r Resource) newMainDeployment(obj providerv1alpha1.AzureConfig) (Deployment, error) {
```

Comment:
Renamed to obj. `cluster` was weird. In general I favour `obj` over `customOjbect`. It's shorter and we use so often that we can reserve that name excessively for that.

Reply:
FYI I saw in the k8s community the working seems to be `CRD` and `CR`.

Reply:
Not sure if you aren't confusing things here. My understanding is:

- `CRD` is a resource (/customresourcedefinitions endpoint)
- `CR` is an object of the CRD resource (this object is reconciled by the `CRD` controller and new resource is created with new endpoint, e.g. /azureconifgs); the object can be obtained with `kubectl get customresourcedefinitions cr_name`.
- `obj` is an object of the new resource (/azureconfigs endpoint), created by reconciling the `CR`; the object can be obtained with `kubectl get azureconfigs obj_name`.

Reply:
For me `obj` and `CR` are the same here.